UI/window-dock-youtube-app.cpp

```diff
@@ -0,0 +1,383 @@
+		obs_service_t *service_obj = OBSBasic::Get()->GetService();
+		OBSDataAutoRelease settings =
+			obs_service_get_settings(service_obj);
+		if (IsUserSignedIntoYT()) {
+			channelId.clear(); // renew channelId
+			UpdateChannelId();
+			const char *broadcast_id =
+				obs_data_get_string(settings, "broadcast_id");
+			SetInitEvent(YTSM_ACCOUNT, INGESTION_STARTED,
+				     broadcast_id,
+				     channelId.toStdString().c_str());
+		} else {
+			const char *stream_key =
+				obs_data_get_string(settings, "key");
+			SetInitEvent(YTSM_STREAM_KEY, INGESTION_STARTED,
+				     stream_key);
+		}
+	} else {
+		SetInitEvent(IsUserSignedIntoYT() ? YTSM_ACCOUNT
+						  : YTSM_STREAM_KEY);
+	}
+
+	dockBrowser->reloadPage();
+}
+
+void YouTubeAppDock::UpdateChannelId()
+{
+	if (channelId.isEmpty()) {
+		YoutubeApiWrappers *apiYouTube = GetYTApi();
+		if (apiYouTube) {
+			ChannelDescription channel_description;
+			if (apiYouTube->GetChannelDescription(
+				    channel_description)) {
+				channelId = channel_description.id;
+			} else {
+				blog(LOG_ERROR,
+				     "YT: AccountConnected() Failed to get channel id");
+			}
+		}
+	}
+}
+
+void YouTubeAppDock::SetInitEvent(streaming_mode_t mode, const char *event,
+				  const char *video_id, const char *channelId)
+{
+	const std::string version = App()->GetVersionString();
+	blog(LOG_DEBUG, "YT: obs version: %s", version.c_str());
+
+	QString api_event;
+	if (event) {
+		if (mode == YTSM_ACCOUNT) {
+			api_event = QString(R"""(,
+					initEvent: {
+						type: '%1',
+						channelId: '%2',
+						videoId: '%3',
+					}
+			)""")
+					    .arg(event)
+					    .arg(channelId)
+					    .arg(video_id);
+		} else {
+			api_event = QString(R"""(,
+					initEvent: {
+						type: '%1',
+						streamKey: '%2',
+					}
+			)""")
+					    .arg(event)
+					    .arg(video_id);
+		}
+	}
+
+	std::string script = QString(R"""(
+		let obs_name = '%1';
+		let obs_version = '%2';
+		let client_mode = %3;
+		if (window.location.hostname == 'studio.youtube.com') {
+			console.log("name:", obs_name);
+			console.log("version:", obs_version);
+			console.log("initEvent:", {
+					initClientMode: client_mode
+					%4 });
+			if (window.ytlsapi && window.ytlsapi.init)
+				window.ytlsapi.init(obs_name, obs_version, undefined, {
+					initClientMode: client_mode
+					%4 });
+		}
+	)""")
+				     .arg("OBS")
+				     .arg(version.c_str())
+				     .arg(mode == YTSM_ACCOUNT ? "'ACCOUNT'"
+							       : "'STREAM_KEY'")
+				     .arg(api_event)
+				     .toStdString();
+	dockBrowser->setStartupScript(script);
+}
+
+YoutubeApiWrappers *YouTubeAppDock::GetYTApi()
+{
+	Auth *auth = OBSBasic::Get()->GetAuth();
+	if (auth) {
+		YoutubeApiWrappers *apiYouTube(
+			dynamic_cast<YoutubeApiWrappers *>(auth));
+		if (apiYouTube) {
+			return apiYouTube;
+		} else {
+			blog(LOG_ERROR,
+			     "YT: GetYTApi() Failed to get YoutubeApiWrappers");
+		}
+	} else
+		blog(LOG_ERROR, "YT: GetYTApi() Failed to get Auth");
```

Comment:
If you enclose one branch of an `if` statement, then all branches must be enclosed.

Reply:
Done.